### PR TITLE
[@sap/xsenv] Add types for @sap/xsenv module

### PR DESCRIPTION
**Pull request by mad-mike**
**The module is available on https://npm.sap.com and SAP Service Marketplace**

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
@mad-mike Please provide the documentation. There is no package with @sap/xsenv.
https://www.npmjs.com/search?q=%40sap%2Fxsenv
If your package is not public you want to include definitions in the package instead of on definitely typed

**Comment by mad-mike** [comment]
Hi @sheetalkamat,
The package is public, but it's not listed on npmjs.com. As I wrote in the first line of the pull request and in the header of the d.ts: this package is available on https://npm.sap.com.
Try `npm search xsenv --registry https://npm.sap.com`.

**Comment by sheetalkamat** [comment]
@andy-ms Can you please suggest if these definitions can be here or need to be included in the package itself.

**Comment by andy-ms** [comment]
`npm search` works, but `npm info xsenv --registry https://npm.sap.com` is giving me a 404 error, so I can't get much info about this project. The `// Project:` link should be to a URL viewable in a web browser.
Are you the package author? If so, it would be better to [bundle](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) the types.

**Comment by mad-mike** [comment]
Hi @andy-ms,
the module is called `@sap/xsenv`. I only used the short name for search.
I am not the package author and cannot add to the project. This is a module provided by [SAP](www.sap.com) to simplify programming on their HANA XSA platform. This is not a classical open source project in the sense that everyone can contribute, but the source code is distributed freely via the SAP registry.
I could change the `// Project` link to the [HANA platform documentation](https://help.sap.com/viewer/p/SAP_HANA_PLATFORM), if this helps to categorize the module.

File: types/sap__xsenv/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// Type definitions for @sap/xsenv 1.2
+// Project: https://npm.sap.com/@sap/xsenv
+// Definitions by: [Michael Müller] <https://github.com/mad-mike>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/* tslint:disable no-single-declare-module
```
Comment (andy-ms) [inline types/sap__xsenv/index.d.ts]: Actually, it should be possible to do away with the `declare module`. You just need a path mapping like `google-cloud__storage` has:
```json
"paths":{
  "@sap/xsanv": ["sap__xsenv"]
},
```

**Comment by andy-ms** [comment]
OK. Could you include a comment near the top about how this project requires the `--registry` option?
Also, the documentation link is better than an NPM registry URL, but it would be better to have something that describes this particular package itself, rather than the platform as a whole.

File: types/sap__xsenv/index.d.ts (hunk 1 above)
Comment (mad-mike) [inline types/sap__xsenv/index.d.ts]: Thanks, I didn't know that technique!

**Comment by mad-mike** [comment]
Hi @andy-ms,
I've improved the types as recommended.
Unfortunately there is no stable link to the detailed documentation, therefore I added a few comments to the header of the definition file.
Hope everything is ok now!

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**